qiita_pet/handlers/study_handlers/tests/test_artifact.py

```diff
@@ -126,16 +127,19 @@ def test_post_artifact(self):
         self.assertEqual(response.code, 200)
 
         # make sure new artifact created
-        obs = r_client.get('prep_template_%s' % self.prep.id)
+        obs = r_client.get('prep_template_%d' % self.prep.id)
         self.assertIsNotNone(obs)
-        redis_info = loads(r_client.get(loads(obs)['job_id']))
-        while redis_info['status_msg'] == 'Running':
-            sleep(0.05)
-            redis_info = loads(r_client.get(loads(obs)['job_id']))
-        new_artifact_id = get_count('qiita.artifact')
-        artifact = Artifact(new_artifact_id)
-        self.assertEqual(artifact.name, 'New Artifact Handler test')
-        self._files_to_remove.extend([fp for _, fp, _ in artifact.filepaths])
+        payload = loads(obs)
+        job_id = payload['job_id']
+        if payload['is_qiita_job']:
+            job = ProcessingJob(job_id)
+            while job.status not in ('success', 'error'):
+                sleep(0.05)
+        else:
+            redis_info = loads(r_client.get(job_id))
+            while redis_info['status_msg'] == 'Running':
+                sleep(0.05)
+                redis_info = loads(r_client.get(job_id))
 
```

Comment:
I'd build up a library of testing functions in a separate python script to store these sorts of functions.

But its definitely not blocking - if you guys think this is a good idea, I guess this can be raised as a issue to refactor later.